src/providers/files/files_ops.c

```diff
@@ -272,13 +274,96 @@ static errno_t delete_all_users(struct sss_domain_info *dom)
     }
 
     ret = EOK;
+    DEBUG(SSSDBG_TRACE_FUNC, "All users deleted from database\n");
 
 done:
     talloc_free(tmp_ctx);
 
     return ret;
 }
 
+static errno_t delete_non_existing_users(struct files_id_ctx *id_ctx)
+{
+    TALLOC_CTX *tmp_ctx = NULL;
+    TALLOC_CTX *tmp_ctx_users = NULL;
+    TALLOC_CTX *tmp_ctx_whitelist = NULL;
+    struct ldb_dn *base_dn;
+    struct passwd **users = NULL;
+    const char **whitelist = NULL;
+    int users_num = 0;
+    errno_t ret;
+
+    tmp_ctx_users = talloc_new(NULL);
+    if (tmp_ctx_users == NULL) {
+        DEBUG(SSSDBG_FATAL_FAILURE, "Out of memory!\n");
+        return ENOMEM;
+    }
+
+    /* Process password files to get existing users */
+    for (size_t i = 0; id_ctx->passwd_files[i] != NULL; i++) {
+        ret = enum_files_users(tmp_ctx_users,
+                               id_ctx->passwd_files[i],
+                               &users);
+        if (ret != EOK) {
+            goto done;
+        }
+    }
+
+    tmp_ctx_whitelist = talloc_new(NULL);
```

Comment:
Is there a reason to create new temp talloc context instead of reusing the same?